example/decl-appl-bnf.eag

```diff
@@ -0,0 +1,38 @@
+// declarations/applications: a reduced example in BNF
+// see decl-appl.eag for an EBNF specification
+// 
+// build generator: dub build.
+// run generator: ./gamma -s example/decl-appl-bnf.eag
+// run generated compiler: ./DeclAppl test/decl-appl/test1
+// output: aba ; ab ;
+
+Table = | id ";" Table .
+
+DeclAppl <+ Table>:
+    Statements<, Table>
+.
+
+Statements<- Table, + Table1>:
+        "DECL" ident<id> Statements<id ";" Table, Table1>.
+Statements<- Table, + Table1>:
+        "APPL" ident<id> Find<id, Table> Statements<Table, Table1>.
+Statements <- Table, + Table>: .
+
+x = "a" | "b".
+id* = x | id x.
+
+x<+ "a": x>: "a".
+x<+ "b": x>: "b".
+
+ident*<+ id>:
```

Comment:
Do you really need this change?
One level down, is's meta `x` as well as hyper `x`.

Better add a sentence to the tutorial that such a "lifting" is very common for "terminal symbols" (non-terminals with `*`).

Reply:
For newbies which are not familiar with concepts and which part of specification is representing which concept it is very hard to follow a describing text if there are equal named identifiers with different meanings.

It also gets very hard to reference them in the tutorial text right: As concepts are not clear yet for a newbie, the reference "meta-nonterminal 'id'" or "hyper-nonterminal 'id'" could not be differentiated and the tutorial would need to talk about positions of 'id'.

That's at the end the real reason for having different names for meta and hyper non-terminals. 
I'm looking from a didactic point of view.
Talking about "lifting" would not help a lot here as it is just another concept needing explanation...

Reply:
BTW: it's `ident` / `ident` for Oberon-0, but `id` / `id` for Decl/Appl to make better use of the space on a slide. I thought about `xs` instead of your `id`, but `xs '';' Table` destroys the intuition. Why `xs`? Because the left recursion for `ident` only makes sense if you have letters and digits, but only letters at the beginning. But with only "a" and "b", there is no motivation for `identTail`.

Also from an educational perspective, I don't like "ident" vs. "id". Even as experts, we have no idea whether it should be `id <ident>` or the other way around `ident <id>`. And shall we put the `id` into the table, or the `ident`? Naming both `ident` is the best way to tell: "don't care!" You want to find the (hyper) identifier from APPL in the (meta) table.

(BTW: you somehow avoided with `x` / `x`` in your text by just talking about meta `x`.)

If you're still not convinced: a compromise could be `Variable <ident>`.

Reply:
Referencing it as _affix variable_ `<ident>` in the text is a good idea!

May be it is also a good idea to talk about lifting. 
I remember,  we maybe want to introduce a syntactic shortcut for that...

Reply:
Renamed "ident" back to "id" at examples/bnf/decl-apl.eag and examples/decl-appl.eag (where I also renamed it, previously). Reworked the tutorial text (not in this commit, see wiki).